Sources/SwiftTranslate/Bootstrap/SwiftTranslate.swift

```diff
@@ -67,55 +67,59 @@ struct SwiftTranslate: AsyncParsableCommand {
-            targetLanguages = nil
-        } else if languages.first?.rawValue == "all" {
-            targetLanguages = Set(Language.allCommon)
-        } else {
-            let invalidLanguages = languages.filter({ !Language.allCommon.contains($0) }).map(\.rawValue)
-            guard invalidLanguages.isEmpty else {
-                throw ValidationError("Invalid language(s) provided: \(invalidLanguages.joined(separator: ", "))")
+    func run() async {
+        do {
+            var translator: TranslationService
+
+            switch service {
+                case .google:
+                    translator = GoogleTranslator(apiKey: apiToken)
+                case .openAI:
+                    translator = OpenAITranslator(with: apiToken, model: model)
             }
-            targetLanguages = Set(languages)
-        }
-        
-        var mode: TranslationCoordinator.Mode
-        if let text = textOptions.text {
-            guard let targetLanguages else {
-                throw ValidationError("Target language(s) is required for text translation")
+
+            var targetLanguages: Set<Language>?
+            if languages.first?.rawValue == "__in_catalog" {
+                targetLanguages = nil
+            } else if languages.first?.rawValue == "all" {
+                targetLanguages = Set(Language.allCommon)
+            } else {
+                let invalidLanguages = languages.filter({ !Language.allCommon.contains($0) }).map(\.rawValue)
+                guard invalidLanguages.isEmpty else {
+                    throw ValidationError("Invalid language(s) provided: \(invalidLanguages.joined(separator: ", "))")
+                }
+                targetLanguages = Set(languages)
             }
-            mode = .text(text, targetLanguages)
-        } else if let fileOrDirectory = catalogOptions.fileOrDirectory.first {
-            if let unwrappedTargetLanguages = targetLanguages, !unwrappedTargetLanguages.contains(.english) {
-                targetLanguages?.insert(.english)
+
+            var mode: TranslationCoordinator.Mode
+            if let text = textOptions.text {
+                guard let targetLanguages else {
+                    throw ValidationError("Target language(s) is required for text translation")
+                }
+                mode = .text(text, targetLanguages)
+            } else if let fileOrDirectory = catalogOptions.fileOrDirectory.first {
+                if let unwrappedTargetLanguages = targetLanguages, !unwrappedTargetLanguages.contains(.english) {
+                    targetLanguages?.insert(.english)
+                }
+                mode = .fileOrDirectory(
+                    URL(fileURLWithPath: fileOrDirectory),
+                    targetLanguages,
+                    overwrite: catalogOptions.overwriteExisting
+                )
+            } else {
+                throw ValidationError("No text or string catalog file to translate provided")
```

Comment:
I think this change would silence these errors? Unless I'm missing something

Reply:
I just wrapped the whole `run` function into a `do-catch` so that we can print the errors in red.
Since this `throw` is inside the `do`, it will be caught from the `catch` at line 126 and printed as a red error (line 127)

Reply:
Yea, but this removes the built in functionality of `swift-argument-parser` to surface hints about accepted commands / parameters with the error

Reply:
<img width="855" alt="Screenshot 2025-03-15 at 18 30 31" src="https://github.com/user-attachments/assets/43c25fbc-c40e-4867-afd4-8c0c5f8f2e29" />
That will be available anyway, I tried to use a random `--asd` option in the screenshot.

Reply:
Maybe I got your point. What if we add `throw error` on line 127 after the `print("\(error)".red)`?
In this way, the error will be propagated to the `run()` caller.